### PR TITLE
Fix borderless fullscreen unexpectedly running in exclusive mode on windows

### PR DESCRIPTION
**Pull request by peppy**
This employs the age-old hack that we've been relying on in stable/osuTK of misaligning the window with the screen. On a quick check, many other games use this same method, so it seems like the best we're going to do for the time being.

I've left the existing borderless operation mode for non-windows platforms, since it works better.

Closes https://github.com/ppy/osu-framework/issues/4068
Addresses part of https://github.com/ppy/osu-framework/issues/4074#issuecomment-739690107
Brings back issue mentioned in https://github.com/ppy/osu-framework/issues/3456 (i believe we may be able to solve this, but this is a separate task which requires further investigation into how other games/windowing systems get it working).

### CODE REVIEW

File: osu.Framework/Platform/SDL2DesktopWindow.cs [hunk 1]
```diff
@@ -962,6 +962,7 @@ private void updateWindowStateAndSize()
                 case WindowState.Normal:
                     Size = sizeWindowed.Value;
 
+                    SDL.SDL_SetWindowBordered(SDLWindowHandle, SDL.SDL_bool.SDL_TRUE);
```
Comment (peppy) [inline osu.Framework/Platform/SDL2DesktopWindow.cs:965]: I realise that this should probably be in the windows-only implementation, but it seems to be a harmless noop so left it here for simplicity. Can abstract out if there's an issue with this.

**Comment by Joehuu** [comment]
This seems to update `WindowedSize` to my screen resolution + 1x1 when switching between modes 99% of the time, making windowed like borderless and impossible to resize.

**Comment by peppy** [comment]
Yeah, I've seen this happen more than once, but I can also seem to repro that on master at random. Maybe easier with this change due to there no longer being a huge lag period when changing to borderless?

Was planning on addressing it as a separate issue, but maybe it's best to fix here..

**Review by smoogipoo — APPROVED** [review]

**Comment by frenzibyte** [comment]
This is giving me borderless unresizable windows once switching back to windowed:

**Comment by peppy** [comment]
Sure it's not something to do with maximised state or otherwise? can you set a breakpoint and see why the border restoration call isn't being hit?

**Pull request merged**